Userland/Libraries/LibWeb/HTML/FormAssociatedElement.cpp

```diff
@@ -589,4 +586,240 @@ void FormAssociatedTextControlElement::set_the_selection_range(Optional<WebIDL::
     }
 }
 
+void FormAssociatedTextControlElement::handle_insert(String const& data)
+{
+    auto text_node = form_associated_element_to_text_node();
+    if (!text_node || !text_node->is_editable())
+        return;
+
+    String data_for_insertion = data;
+    if (auto max_length = text_node->max_length(); max_length.has_value()) {
+        auto remaining_length = *max_length - text_node->data().code_points().length();
```

Comment:
This used to have a FIXME about using UTF-16 code unit length, let's not drop that.

In the mean time, I'm not sure that code point length is a step in the right direction here. `substring_from_byte_offset` below is byte-based, not code point-based. That must be crashy on non-ASCII text?